### PR TITLE
feat(list): added icons, plain, bordered

### PR DESCRIPTION
**Pull request by mcoker**
**What**: Closes https://github.com/patternfly/patternfly-react/issues/5336

### CODE REVIEW

File: packages/react-core/src/components/List/List.tsx [hunk 1]
```diff
@@ -26,6 +26,12 @@ export interface ListProps extends Omit<React.HTMLProps<HTMLUListElement | HTMLO
   className?: string;
   /** Adds list variant styles */
   variant?: ListVariant.inline;
+  /** Modifies the list to add borders between items */
+  isBordered?: boolean;
+  /** Modifies the list to include plain styling */
+  isPlain?: boolean;
+  /** Modifies the list to include plain styling */
```
Comment (evwilkin) [inline packages/react-core/src/components/List/List.tsx:33]: it looks like this description needs to be updated

**Comment by evwilkin** [comment]
@mcoker more of a Core question but the [designs](https://marvelapp.com/prototype/7gc36d1/screen/75109901) seem to show icons & text vertically centered but here they look bottom-aligned?

**Comment by mcoker** [comment]
@evwilkin made an update that should help the alignment.

**Review by evwilkin — APPROVED** [review]
LGTM - thanks!

**Review by mcarrano — APPROVED** [review]
Looks great. Thanks @mcoker !

**Review by redallen — APPROVED** [review]

**Review by nicolethoen — APPROVED** [review]

**Pull request merged**